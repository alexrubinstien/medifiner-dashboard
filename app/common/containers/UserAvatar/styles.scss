@import './app/common/styles/utils/_vars.scss';

.user-avatar {
  width: 54px;
  height: 54px;
  background: $primary-color;
  border-radius: 50%;

  &__text {
    margin: 0;
    padding: 18px 0 20px;
    color: #fff;
    font-size: 16px;
    font-weight: 700;
    line-height: 16px;
    letter-spacing: 1.6px;
    text-align: center;
  }
}
