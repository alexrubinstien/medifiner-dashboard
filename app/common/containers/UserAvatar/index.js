import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import Usermenu from '@common/components/Usermenu';

import authSelectors from '@selectors/auth';

import './styles.scss';

const UserAvatar = ({ me }) => (
  <div className="user-avatar">
    <div className="user-avatar__text">
      {me && (
        <Usermenu
          label={(me.firstName && me.lastName) ? (me.firstName[0] + me.lastName[0]) : null}
          items={[
              // {
              //   href: '/account-settings',
              //   text: 'Account Settings',
              // },
              {
                href: '/logout',
                text: 'Logout',
              }]}
        />
      )}
    </div>
  </div>
);

const mapStateToProps = state => ({
  me: authSelectors.getMe(state),
});

UserAvatar.propTypes = {
  me: PropTypes.shape({
    firstName: PropTypes.string,
    lastName: PropTypes.string,
  }).isRequired,
};

export default connect(mapStateToProps)(UserAvatar);
