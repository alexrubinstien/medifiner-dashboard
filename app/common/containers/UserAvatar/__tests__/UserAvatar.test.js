/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import UserAvatar from '../index.js';

describe('With Snapshot Testing', () => {
  it('User avatar component', () => {
    const mockStore = configureStore([]);
    const initialState = {
      auth: {
        me: {
          data: {
            firstName: 'John',
            lastName: 'Doe',
          },
        },
      },
    };
    const store = mockStore(initialState);

    const component = renderer.create(
      <Provider store={store}>
        <UserAvatar />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
