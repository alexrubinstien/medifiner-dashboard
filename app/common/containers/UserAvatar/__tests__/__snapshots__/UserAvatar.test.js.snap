// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing User avatar component 1`] = `
<div
  className="user-avatar"
>
  <p
    className="user-avatar__text"
  >
    J
    D
  </p>
</div>
`;
